### ISSUE TITLE
The "Sketch" toolbar button should always trigger Start Sketch, never Edit Sketch

### ISSUE THREAD

**Issue by franknoirot** [KittyCAD/modeling-app#4273]
Currently we toggle the meaning of the "Sketch" button (and its label) depending on the user's selection. If they have a selection that resolves to a sketch, we show the edit sketch version of the button.

As an investor pointed out during the KittyCAMP 2024 makeathon, this is confusing for existing CAD users, who are used to being able to select a face before or after hitting the sketch button and have it trigger a new sketch on face.

We need a better UX for editing existing sketches before we can get rid of this flow, so this is blocked until we have a feature tree, which would be the normal way of editing a sketch (either by double-clicking or via right-click menu).

**Comment by pierremtb** [comment]
Blocked by https://github.com/orgs/KittyCAD/projects/35/views/1?layout=board&verticalGroupedBy%5BcolumnId%5D=145912631&pane=issue&itemId=86547834&issue=KittyCAD%7Cmodeling-app%7C4274

**Comment by max-mrgrsk** [comment]
How would you edit sketch in the kcl mode? Same applies for other buttons (fillet, chamfer etc)

**Comment by franknoirot** [comment]
> How would you edit sketch in the kcl mode

Sorry @max-mrgrsk what do you mean here? I believe you would just edit the KCL directly, which you can do today? I'm not quite sure what the KCL "mode" you're speaking of is.

**Comment by max-mrgrsk** [comment]
@franknoirot , what I meant is that currently, we have two ways to edit a sketch:

1. Editing the KCL code directly.
2. Using the “Edit Sketch” functionality via the button, which takes you back to sketching mode where you can continue making changes graphically.

If we remove the “Edit” functionality from the Sketch button, the second option (returning to sketching mode to graphically edit) will no longer be available. This might make things trickier for users who prefer or rely on the graphical interface for editing.

**Comment by franknoirot** [comment]
> If we remove the “Edit” functionality from the Sketch button, the second option (returning to sketching mode to graphically edit) will no longer be available. This might make things trickier for users who prefer or rely on the graphical interface for editing.

Totally, you're right @max-mrgrsk. That's why I included this note in the description:

> ...so this is blocked until we have a feature tree, which would be the normal way of editing a sketch (either by double-clicking or via right-click menu).

Does that seem alright? The feature tree would become the primary graphical interface way to edit a sketch: either by double-clicking the Sketch in the feature tree or `right-click > Edit sketch`. Does that seem reasonable? That will be unblocked with #5497, so if that seems good we might be able to make a PR to close this after that lands.

**Comment by franknoirot** [comment]
#5497 has merged, so this is now unblocked.

**Comment by franknoirot** [comment]
Here is Onshape's toolbar:

![Onshape's "sketch" button](https://github.com/user-attachments/assets/f01eaecf-7e2a-44f9-aa3e-998d18c6c839)

And here is Fusion's:

![Fusion's "create sketch" button](https://github.com/user-attachments/assets/f3ae8bce-b636-43b9-b042-8dc3873d5e4b)

Both have that button mean "Create a new sketch", and both have their edit flows through their timeline/feature tree and through a right-click menu in the scene:

![Fusion's timeline right-click menu item](https://github.com/user-attachments/assets/8ccbca54-5500-4a85-b1eb-a958fde08e0b)
![Fusion's scene right-click menu item](https://github.com/user-attachments/assets/4aa3387a-1bdb-4fcc-b2d3-f403a027881b)

It would be unusual to have either two buttons for sketch mode or to have it do the dual job it does today. We do not have the selection logic in place to have an in-scene right-click menu item, but in the interim I think it's important that we have the toolbar do something consistent and intuitive, and only add new sketches in the same manner that every other toolbar button only ever _adds_ new geometry.

**Comment by max-mrgrsk** [comment]
Yeah, we have a harder job than they do. **They only have the feature tree**, while **we have both**: the tree and the code.

So **we need a solution that works for both** scenarios.

The current approach works for both, which is great. But it’s not ideal because you might accidentally go into edit mode instead of creating a new sketch.

However, removing this option without providing an alternative will make the code-based design process worse. You can’t design without editing sketches, and you won’t want to open and close the feature tree every time you need to edit something. That means you just wouldn’t use the code pane at all. My concern is that this change could effectively kill the code-to-CAD workflow in our app.

One possible solution:

- Make the sketch button **only start sketch** mode **for feature tree users**.
- For **code users**, **allow swapping** back and forth as needed.

**Comment by franknoirot** [comment]
I don't think that a clearcut "feature tree users" and "code users" distinction will exist for us to trigger major UI changes on: users can act as one or the other pretty fluidly.

We don't have any mechanism to track through which method something was selected, and though I can't articulate why exactly I am hesitant to start tracking that sort of thing. I believe it is a strength that selections be agnostic of which way you make them: if anything I think we still have too much special logic between the selection workflows.

**Comment by franknoirot** [comment]
Huh I actually enjoy this approach quite a lot! Want to try it out for yourself @max-mrgrsk? https://github.com/KittyCAD/modeling-app/pull/5691

## Demo

https://github.com/user-attachments/assets/cd35bc4c-6515-41f3-9467-72839b04dd56

**Comment by max-mrgrsk** [comment]
I love it!

**Comment by pierremtb** [comment]
@max-mrgrsk @franknoirot I think this is a great solution at least for the time being!
